### PR TITLE
Fix restart with auto-expand replicas

### PR DESCRIPTION
**Pull request by henningandersen**
No longer contract `number_of_replicas` when a node is marked
restarting, since this poses a risk of the shard/index becoming
unavailable.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeShutdownAllocationDecider.java [hunk 1]
```diff
@@ -76,14 +76,14 @@ public Decision shouldAutoExpandToNode(IndexMetadata indexMetadata, DiscoveryNod
         SingleNodeShutdownMetadata thisNodeShutdownMetadata = getNodeShutdownMetadata(allocation.metadata(), node.getId());
 
         if (thisNodeShutdownMetadata == null) {
-            return allocation.decision(Decision.YES, NAME, "node [%s] is not preparing for removal from the cluster");
+            return allocation.decision(Decision.YES, NAME, "node [%s] is not preparing for removal from the cluster", node.getId());
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeShutdownAllocationDecider.java:79]: Just spotted this and fixed it in this PR too (as well as added the test for it).

**Comment by gwbrown** [comment]
I've requested a review from @pgomulka as he's taken over Node Shutdown on the core/infra side of things. I'll take a look as well, but he should be the primary contact going forward.

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeShutdownAllocationDecider.java (hunk 1 above)
Comment (gwbrown) [reply server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeShutdownAllocationDecider.java:79]: Excellent catch, thanks very much for the fix.

**Review by AthenaEryma — APPROVED** [review]

**Review by dakrone — APPROVED** [review]
LGTM, thanks Henning

**Comment by henningandersen** [comment]
@elastichmachine run elasticsearch-ci/rest-compatibility

**Review by pgomulka — COMMENTED** [review]
LGTM

**Pull request merged**